packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -75,11 +75,10 @@ class _DefaultDoctorValidatorsProvider implements DoctorValidatorsProvider {
         if (webWorkflow.appliesToHostPlatform)
           const WebValidator(),
         // Add desktop doctors to workflow if the flag is enabled.
-        if (flutterDesktopEnabled)
-          ...<DoctorValidator>[
-            if (linuxWorkflow.appliesToHostPlatform) LinuxDoctorValidator(),
-            if (windowsWorkflow.appliesToHostPlatform) visualStudioValidator,
-          ],
+        if (featureFlags.isLinuxEnabled && linuxWorkflow.appliesToHostPlatform)
```

Comment:
Linux and Windows are behind a check here, but macOS isn't; it should be consistent (even though in practice it doesn't matter, since iOS applying is the same as being on macOS). Per my later comment I think the better option is to remove them all here, but if you don't go that route you should add the check for macOS.

Reply:
Done